kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/KafkaRequestHandler.java

```diff
@@ -2044,7 +2044,7 @@ protected boolean isTransactionTopic(String topic) {
         CompletableFuture<PartitionMetadata> returnFuture = new CompletableFuture<>();
 
         topicManager.getTopicBroker(topic.toString())
-            .thenCompose(pair -> getProtocolDataToAdvertise(pair, topic))
+            .thenCompose(address -> getProtocolDataToAdvertise(address, topic))
```

Comment:
can we add a judgement in `getProtocolDataToAdvertise`? if `kafkaListenerName` is used, do not convert broker url to kafka advertised address in `getProtocolDataToAdvertise`, just use the `address` here combine with `SecurityProtocol`.

Reply:
Could you explain why not use `getProtocolDataToAdvertise` if `kafkaListenerName` is used? For example, in `KafkaListenerNameTest`, the `address` here is `localhost:15002` and `getProtocolDataToAdvertise(address, topic)` will be completed with `PLAINTEXT://localhost:15003,SSL://localhost:15004`.

If we don't use `getProtocolDataToAdvertise`, which `SecurityProtocol` should we use? Could you give an example implementation and tell the reason?

Reply:
I think the configuration in `KafkaListenerNameTest` is not correct. We should not set `InternalListenerName` same with `KafkaListenerName`. Broker will use the address in `InternalListenerName` as advertised address and regist to zk path `/loadbalance/brokers`.
If `InternalListenerName` is not set, default the first listener `internal` will be regist to zk path and `getProtocolDataToAdvertise` will failed with "No node for broker under zk://loadbalance"

Reply:
Could you give an example config in test environment? Since there's no proxy, broker can only listen on `localhost:<brokerPort>`, other listeners are unused.

Reply:
Do you mean given following config

```
internalListenerName=internal
advertisedListeners=internal:pulsar://localhost:6650,external:PLAINTEXT://localhost:9092
kafkaListenerName=external
```

The returned address (`stringOptional` is following code) should be `localhost:9092`? 

```java
        topicManager.getTopicBroker(topic.toString())
            .thenCompose(address -> getProtocolDataToAdvertise(address, topic))
            .whenComplete((stringOptional, throwable) -> {
````

Reply:
@wangjialing218 To correct my previous comment, the `advertisedListeners` cannot accept protocols other than `pulsar` or `pulsar+ssl`. Therefore, the `internal:pulsar://localhost:6650,external:PLAINTEXT://localhost:9092` is invalid.

If I didn't understand wrong, maybe you want to configure

```bash
advertisedListeners=pulsar:pulsar://localhost:6650,kop:pulsar://localhost:9092
```

and return the `PLAINTEXT://localhost:9092` as the listeners that will be passed to `EndPoint.getSslEndPoint` or `EndPoint.getPlainTextEndPoint`?

If so, I think you can open a new PR that is based on this PR. Because it introduced a new mapping rule from Pulsar's `TopicDomain` to Kafka's `SecurityProtocol`:
- `pulsar` -> `PLAINTEXT`
- `pulsar+ssl` -> `SSL`
- `SASL_PLAINTEXT` and `SASL_SSL` are not supported in KoP because we use `saslAllowedMechanisms` config to differ whether the SASL is enabled.

And it should be documented as well. I think this PR's original purpose is to provide a dependent method for topic lookup.

Reply:
I want to configure like this:
 `advertisedListeners=pulsar:pulsar://192.168.0.1:6650,kop:pulsar://10.254.1.1:9092`
192.168.0.1 is private network ip and 10.254.1.1 is public network ip. 
In order to deploy pulsar cluster that can be accessed through both public network private network.
https://github.com/apache/pulsar/wiki/PIP-61:-Advertised-multiple-addresses

Current If I configure like this, `getProtocolDataToAdvertise` will failed.
I will open a new PR after this PR merged.

Reply:
Got it.

Reply:
@wangjialing218 It's merged now, you can start the work.